src/azure/azure-mesh-adapter.ts

```diff
@@ -24,21 +25,68 @@ export class AzureMeshAdapter implements MeshAdapter {
     tenants: MeshTenant[],
     startDate: Date,
     endDate: Date,
+  ): Promise<void> {
+    const config = loadConfig();
+    if (config.azure.parentManagementGroups.length == 0) {
+      log.warning(
+        "Please configure an Azure management group ID under which your Subscriptions are placed. " +
+          "Because of a bug in the Azure API collie can not detect this automatically and with this info cost usage lookups are significantly faster.",
+      );
+      await this.getTenantCostsWithSingleQueries(tenants, startDate, endDate);
+    } else {
+      await this.getTenantCostsWithManagementGroupQuery(
+        config.azure.parentManagementGroups,
+        tenants,
+        startDate,
+        endDate,
+      );
+    }
+  }
+
+  private async getTenantCostsWithManagementGroupQuery(
+    managementGroupIds: string[],
+    tenants: MeshTenant[],
+    startDate: Date,
+    endDate: Date,
   ): Promise<void> {
     // Only work on Azure tenants
     const azureTenants = tenants.filter((t) => isSubscription(t.nativeObj));
+    const from = moment(startDate).format("YYYY-MM-DDT00:00:00");
+    const to = moment(endDate).format("YYYY-MM-DDT23:59:59");
+
+    const costInformations = [];
+    for (const mgmntGroupId of managementGroupIds) {
+      const costInformation = await this.azureCli.getCostInfo(
+        mgmntGroupId,
+        from,
+        to,
+      );
+      costInformations.push(...costInformation);
+    }
+
+    const summedCosts = new Map<string, number>();
+    for (const ci of costInformations) {
+      // TODO manage the currency symbol.
```

Comment:
Regarding the currency; I would just add that to the cost info object. That's also what we do with chargeback statements. It would also be great to then being able to export this information via CSV/Table/JSON etc.

Reply:
I guess there wont never be a cost info with multiple currencies? In such case we probably could just add two different cost infos.